File: frontend/app/api/transactions/infectionTypes/createInfectionTypeTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { CreateInfectionTypeType } from "../../../types/infectionTypeTypes"
import { getInfectionTypesFromApi } from "../../infectionTypes/fetchInfectionTypes"
import {
         normalizeInfectionType,
         toCreateInfectionTypeRequest
       } from "../../../utils/infectionTypeMapper"
import {  } from "../../client/apiClient"

type CreateInfectionTypeTransactionParams = {
                                               infectionType: CreateInfectionTypeType
                                               setInfectionTypes: any
                                               onClose?: () => void
                                            }

export async function createInfectionTypeTransaction({
                                                       infectionType,
                                                       setInfectionTypes,
                                                       onClose
                                                     }: CreateInfectionTypeTransactionParams
                                                   )
{
  console.log("createInfectionTypeTransaction")

  await fetch(
                    `${API_BASE_URL}/infection-types`,
                    {
                      method: "POST",
                      headers: {
                                  "Content-Type":
                                  "application/json"
                                },
                      credentials: "include",
                      body: JSON.stringify(
                                              toCreateInfectionTypeRequest(
                                                                            infectionType
                                                                          )
                                            )
                    }
                  )

  const infectionTypes =
    await getInfectionTypesFromApi()

  setInfectionTypes(
                      infectionTypes.map(
                                          normalizeInfectionType
                                        )
                    )

  if (onClose) {onClose()}
}

File: backend/transactions/infection_types/create_infection_type_transaction.py
from infection_types.add_infection_type import add_infection_type
from schemas.infection_type_schemas import AddInfectionTypeRequest
from supabase import Client

def create_infection_type_transaction(
                                        client:Client,
                                        infection_type: AddInfectionTypeRequest,
                                        hospital_id: str
                                     ):

    print("create_infection_type_transaction")

    return add_infection_type(
                                 client=client,
                                 infection_type=infection_type,
                                hospital_id=hospital_id
                             )